File: services/api/src/riders/rider-profile.ts
export interface RiderProfile {
  readonly id: string;
  readonly authSubject: string;
  readonly displayName: string;
  readonly callsign: string | null;
  readonly homeArea: string | null;
  readonly createdAt: string;
  readonly updatedAt: string;
}

export interface UpsertRiderProfileInput {
  readonly newRiderId: string;
  readonly authSubject: string;
  readonly displayName: string;
  readonly callsign: string | null;
  readonly homeArea: string | null;
}

File: apps/mobile/lib/src/models/rider_profile.dart
class RiderProfile {
  const RiderProfile({
    required this.id,
    required this.displayName,
    required this.callsign,
    required this.homeArea,
  });

  final String id;
  final String displayName;
  final String? callsign;
  final String? homeArea;

  factory RiderProfile.fromJson(Map<String, Object?> json) {
    return RiderProfile(
      id: json['id'] as String,
      displayName: json['displayName'] as String,
      callsign: json['callsign'] as String?,
      homeArea: json['homeArea'] as String?,
    );
  }
}

class RiderProfileInput {
  const RiderProfileInput({
    required this.displayName,
    this.callsign,
    this.homeArea,
  });

  final String displayName;
  final String? callsign;
  final String? homeArea;

  Map<String, Object?> toJson() {
    return <String, Object?>{
      'displayName': displayName,
      'callsign': callsign,
      'homeArea': homeArea,
    };
  }
}
